src/adapter/etl-adapter-csv/tests/Flow/ETL/Adapter/CSV/Tests/Integration/CSVExtractorTest.php

```diff
@@ -99,6 +99,64 @@ public function test_extracting_csv_empty_headers() : void
         );
     }
 
+    public function test_extracting_csv_with_utf8_bom() : void
+    {
+        $extractor = from_csv(
+            __DIR__ . '/../Fixtures/csv_with_bom.csv'
+        );
+
```

Comment:
How can we even be sure that this fixture file starts from BOM? 

The more proper approach would be to put here an assertion that would read first 3 bytes from that fixture file and confirm that they are indeed BOM.